libtenzir/include/tenzir/split_regex.hpp

```diff
@@ -0,0 +1,50 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2025 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "tenzir/chunk.hpp"
+#include "tenzir/generator.hpp"
+
+#include <boost/regex.hpp>
+
+#include <optional>
+#include <string_view>
+
+namespace tenzir {
+
+inline auto split_regex(std::string separator) {
+  return [separator = std::move(separator)](generator<chunk_ptr> input)
+           -> generator<std::optional<std::string_view>> {
+    auto sep = separator;
+    auto buffer = std::string{};
+    auto expr = boost::regex{sep, boost::regex_constants::no_except};
```

Comment:
We should not be reconstructing the regex object every time we call this function.